src/hex_repo.erl

```diff
@@ -164,8 +172,16 @@ get_tarball(Name, Version, Options) ->
 %% Internal functions
 %%====================================================================
 
+make_headers(Options) ->
+    lists:foldl(fun set_header/2, #{}, Options).
+
+set_header({etag, ETag}, Headers) -> maps:put(<<"If-None-Match">>, ETag, Headers);
+set_header(_Option, Headers) -> Headers.
+
 get(Client, URI) ->
```

Comment:
since get/2 is private and we have a good case to pass headers I think we can get rid of it and instead always pass headers explicitly; or remove this indirection altogether and call hex_http:get directly, up to you :) leaving as is is fine too.

Reply:
I think this is a good way to isolate the implementation of the headers and how conditional gets are handled from upstream things that use the library. 

On the other hand the **YAGNI** principle tugs at my mind.

I think we should leave it as is, since we already have an options proplist; we might as well keep using it that way, IMO.

Reply:
yup, let's keep as is, thanks!